ocs_ci/helpers/keyrotation_helper.py

```diff
@@ -0,0 +1,294 @@
+            self.cluster_name_name = constants.DEFAULT_CLUSTERNAME_EXTERNAL_MODE
+
+        self.storagecluster_obj = OCP(
+            resource_name=self.cluster_name,
+            namespace=self.cluster_namespace,
+            kind=self.resource_name,
+        )
+
+    def _exec_oc_cmd(self, cmd, **kwargs):
+        """
+        Executes the given command.
+
+        Args:
+            cmd : command to run.
+
+        Returns:
+            str: The output of the command.
+
+        Raises:
+            CommandFailed: If the command fails.
+        """
+
+        try:
+            cmd_out = self.storagecluster_obj.exec_oc_cmd(cmd, **kwargs)
+        except CommandFailed as ex:
+            log.error(f"Error while executing command {cmd}: {ex}")
+            raise ex
+
+        return cmd_out
+
+    def get_keyrotation_schedule(self):
+        """
+        Retrieves the current key rotation schedule for the storage cluster.
+
+        Returns:
+            str: The key rotation schedule.
+        """
+        cmd = f"get {self.resource_name}  -o jsonpath='{{.items[*].spec.encryption.keyRotation.schedule}}'"
+        schedule = self._exec_oc_cmd(cmd, out_yaml_format=False)
+        log.info(f"Keyrotation schedule set to {schedule} in storagecluster spec.")
+        return schedule
+
+    def set_keyrotation_schedule(self, schedule):
+        """
+        Sets the key rotation schedule for the storage cluster.
+
+        Args:
+            schedule (str): The new key rotation schedule.
+        """
+        param = f'[{{"op":"replace","path":"/spec/encryption/keyRotation/schedule","value":"{schedule}"}}]'
+        self.storagecluster_obj.patch(params=param, format_type="json")
+        self.storagecluster_obj.wait_for_resource(
+            constants.STATUS_READY,
+            self.storagecluster_obj.resource_name,
+            column="PHASE",
+            timeout=180,
+        )
+        self.storagecluster_obj.reload_data()
+        if self.get_keyrotation_schedule() == schedule:
+            log.info(f"Storagecluster keyrotation schedule is set as {schedule}")
+
+    def enable_keyrotation(self):
+        """
+        Enables key rotation for the storage cluster.
+
+        Returns:
+            bool: True if key rotation is enabled, False otherwise.
+        """
+        if self.is_keyrotation_enable():
+            log.info("Keyrotation is Already in Enabled state.")
+            return True
+
+        param = '{"spec":{"encryption":{"keyRotation":{"enable":null}}}}'
+        self.storagecluster_obj.patch(
+            params=param, format_type="merge", resource_name=self.cluster_name
+        )
+        self.storagecluster_obj.wait_for_resource(
+            constants.STATUS_READY,
+            self.storagecluster_obj.resource_name,
+            column="PHASE",
+            timeout=180,
+        )
+        self.storagecluster_obj.reload_data()
+        log.info("Keyrotation is enabled in storegeclujster object.")
+        return True
```

Comment:
the function will never return False. How about catching the exception in case `wait_for_resource()` fails and return False?

Reply:
`wait_for_resource()` is returning a bool , so I have added a check: if `wait_for_resource` is false, then return False.